core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/ExpansionGreedyRegionPlanGenerator.scala

```diff
@@ -100,7 +104,19 @@ class ExpansionGreedyRegionPlanGenerator(
             physicalPlan.getUpstreamPhysicalLinkIds(operatorId) ++ physicalPlan
               .getDownstreamPhysicalLinkIds(operatorId)
           })
-          Region(RegionIdentity((index + 1).toString), operatorIds, linkIds)
+          val region = Region(RegionIdentity((index + 1).toString), operatorIds, linkIds)
+
+          val resourceAllocation = new ResourceAllocation(region)
+          val executionClusterInfo = new ExecutionClusterInfo()
+          val (workers, _) = resourceAllocation.allocation(physicalPlan, executionClusterInfo)
+
+          region.physicalOpIds.zip(workers).foreach {
+            case (opId, workerNum) =>
+              val operator = physicalPlan.getOperator(opId)
+              operator.withSuggestedWorkerNum(workerNum)
```

Comment:
by calling `withXXX` function, you are creating a new copy of the physical operator with the new information (e.g., the suggested worker number in this case). this suggestedWorkerNum is supposed to be an input from user to result in changes in RegionConfig, and is not supposed to be changed by the compiler. so please do not create new physical operator with this information changed.